fabric-convention-tags-v2/src/main/java/net/fabricmc/fabric/api/tag/convention/v2/ConventionalBiomeTags.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.tag.convention.v2;
+
+import net.minecraft.registry.tag.TagKey;
+import net.minecraft.world.biome.Biome;
+
+import net.fabricmc.fabric.impl.tag.convention.v2.TagRegistration;
+
+/**
+ * See {@link net.minecraft.registry.tag.BiomeTags} for vanilla tags.
+ * Note that addition to some vanilla tags implies having certain functionality,
+ * and as such certain biome tags exist to mirror vanilla tags, and should be preferred
+ * over vanilla unless its behavior is desired.
+ */
+public final class ConventionalBiomeTags {
+	private ConventionalBiomeTags() {
+	}
+
+	/**
+	 * Biomes that spawn in the Overworld.
+	 * (This is for people who want to tag their biomes as Overworld without getting
+	 * side effects from {@link net.minecraft.registry.tag.BiomeTags.IS_OVERWORLD})
```

Comment:
Nit: What are these side affects? Would it be worth mentioning them to help people decide what tag is right?

Reply:
The side effects is mainly structures. The vanilla structures uses a has_structures biome tag which then often references other biome tags. So if you tag your biome as `minecraft:is_badlands` biome, you get the surface Badlands Mineshafts even if you don't want them. To help prevent these side effects, we have copies of the vanilla biome tags so people can mark what their biome is without being forced to also have certain vanilla content shoved in their biomes. Not sure how to word it in the comment but lmk if you have an idea for how it can be mentioned.

![image](https://github.com/FabricMC/fabric/assets/40846040/41d467ef-880e-42df-9ce3-7a49a1b75fd6)

https://minecraft.wiki/w/Tag#Biomes

Reply:
Aren't structures the main effect, rather than a side-effect? Should modded structures use the `c` tags in their placement tags? If so, should vanilla structures be different? If you don't want the vanilla badlands mineshafts to generate in your badlands biome, shouldn't you rather modify the `#minecraft:has_structure/mineshaft_mesa` tag?

Also, `#minecraft:is_overworld` is used in `#minecraft:has_structure/stronghold`, and having that tag not include all overworld biomes would cause the probably unwanted side-effect of some strongholds not generating. (The biasing of the stronghold to suitable biomes is done with the `#minecraft:stronghold_biased_to` tag.)

Reply:
@jacobsjo No loader has the ability to *remove* entries from a tag. And tags by default require overriding the entire tag in order to remove an entry which would cause conflicts between multiple mods trying to override the has_structure biome tag, remove the is_badlands tag, and specify each badlands biome manually without their own biome.

It is not a feasible solution. So the solution is a second copied tag that pulls entries from the vanilla tag. And you add your biome to the second tag if you don't want the vanilla biome tag's side effect. The people querying for specific biome tags would use the c tag instead for compat. This gives choice to the biome makers to avoid the vanilla content while still allowing mod compat to be accomplished.

Reply:
That makes sense. I think its important to document that people would most likely want to include their overworld biomes in `#minecraft:has_structure/stronghold` then, even when they don't want the stronghold in their biome.

Reply:
javadoc was updated to mention to add to the has_structure/strongholds tag if omitting from the vanilla is_overworld tag